### DISCUSSION TITLE
Testing XML parsers and the resolver

### DISCUSSION THREAD

**Discussion by carlosame** [css4j/xml-dtd#10] [Ideas]
I suggest the creation of a set of three different entity-preservation tests for any of the places where XML parser and Transformers are created, in code that uses xml-dtd:

1) Entities are replaced in a document with a DTD that has both Public and System ID, like:

```xml
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
```

The test may have a timeout of about 5s, to make sure that no outward connection to `w3.org` is made (`w3.org` throttles DTD connections).

<br/>

2) The DTD has no Public nor System IDs:
```xml
<!DOCTYPE html>
```
That is, an HTML5 DOCTYPE.

<br/>

3) No DTD at all, but entities keep being replaced (entity mapping provided as external subset). This requires an XML parser that supports `EntityResolver2` (all the widely used parsers do).

**Comment by vmassol** [comment]
> I'd suggest that you create a set of three entity-preservation tests for any of the places where XML parser and Transformers are created in XWiki:

Thanks for the suggestion. I've started adding some unit tests in XMLUtils. Will need to add some more in the future.
